src/types/list.rs

```diff
@@ -40,6 +39,17 @@ impl PyList {
         }
     }
 
+    /// Construct a new empty list with the given capacity.
+    pub fn with_capacity<'p>(py: Python<'p>, capacity: usize) -> &'p PyList {
```

Comment:
I think this should probably not be named `with_capacity`, because it actually *initializes the list* rather than reserving a fixed amount of space for it. This will be confusing for people used to `Vec`'s behavior, which distinguishes between size and capacity.

I'm also wondering if we should take the initializer object as an `Option<&PyObject>` parameter.

Reply:
That's a very good idea.
How would you name it?